R4ML/R/ml.model.lm.R

```diff
@@ -338,10 +339,127 @@ predict.r4ml.lm <- function(object, data) {
 
     # orgainze result for presentation to user
     if (testing) {
-      stats <- SparkR::as.data.frame(r4ml.read.csv(statsPath, header=FALSE, stringsAsFactors=FALSE))
-      return(list("predictions"=preds, "statistics"=stats))
+      stats <- SparkR::as.data.frame(r4ml.read.csv(statsPath, header = FALSE,
+                                                   stringsAsFactors = FALSE))
+      return(list("predictions" = preds, "statistics" = stats))
+    } else {
+      return(list("predictions" = preds))
     }
-    else {
-      return(list("predictions"=preds))
+}
+
+r4ml.lm.se <- function(object, seed = 5) {
+  logSource <- "r4ml.lm.se"
+
+  df <- object@training_data
+  training_nrow <- SparkR::nrow(df)
+
+  if (training_nrow > 100000) {
+    # since the dataset has >100k rows we want a sample we can fit on the driver
+    frac <- 100000 / training_nrow # we want a sample of ~50k rows
+    df <- SparkR::sample(df, withReplacement = FALSE, fraction = frac, seed = seed)
+  }
+
+  df <- SparkR::as.data.frame(df)
+
+  colnames(df)[which(colnames(df) == object@labelColumnName)] <- "y"
+
+  if (object@intercept) {
+    fit <- lm(y ~ ., data = df )
+  } else {
+    fit <- lm(y ~ . -1, data = df )
+  }
+
+  coefs <- summary(fit)$coefficients
+  coefs <- as.data.frame(coefs)
+
+  # rescale the std. error
+  coefs$`Std. Error` <- coefs$`Std. Error` * sqrt(nrow(df) / training_nrow)
+
+  coefs$`t value` <- coefs$Estimate / coefs$`Std. Error`
+
+  coefs$`Pr(>|t|)` <- 2 * pt(abs(coefs$`t value`), training_nrow, lower.tail = FALSE)
+
+  return(coefs)
+}
+
+#' @name summary.r4ml.lm
+#' @title LM Summary
+#' @description Summarizes an R4ML LM model \eqn{a + b}
+#' @param object (r4ml.lm): An R4ML LM model
+#' @return a summary of the model
+#' @export
+#' @examples \dontrun{
+#' summary.r4ml.lm(r4ml_lm_model)
+#' }
+summary.r4ml.lm <- function(object) {
+  logSource <- "summary.r4ml.lm"
+  
+  cat("Call:\n")
+  cat(object@call)
+  cat("\n\n")
+
+  cat("Residuals:\n")
+  cat("   Mean: ")
+  cat(round(x = as.numeric(object@dmlOuts$stats["AVG_RES_Y", "value"]), digits = 5))
+  cat("   St. Dev.: ")
+  cat(round(x = as.numeric(object@dmlOuts$stats["STDEV_RES_Y", "value"]), digits = 5))
+  cat("\n\n")
+
+  cat("Coefficients:\n")
+  coef_df <- object@coefficients
+  base::colnames(coef_df) <- c("Estimate")
+  coef_df$`Std. Error` <- NA
+  coef_df$`t value` <- NA
+  coef_df$`Pr(>|t|)` <- NA
+
+  
+ i <- 0
+  while (i < 3) {
+    i <- i + 1
+    se_df <- r4ml.lm.se(object, seed = i)
+
+    ### need to make sure the order is the same ###
+    se_df <- se_df[order(row.names(se_df)), ]
+    coef_df <- coef_df[order(row.names(coef_df)), ]
+
+    ### caculate percent diff between the full and sample betas ###
+    B_full <- coef_df$Estimate
+    B_sub <- se_df$Estimate
+  
+    pd <- abs(B_sub - B_full) / ((B_sub + B_full) / 2)
```

Comment:
@bdwyer2 I would recommend using B_full in the denominator rather than the average of B_full and B_sub.